modules/apps/29-fee/transfer_test.go

```diff
@@ -68,3 +68,74 @@ func (suite *FeeTestSuite) TestFeeTransfer() {
 		fee.AckFee.Add(fee.TimeoutFee...), // ack fee paid, timeout fee refunded
 		sdk.NewCoins(suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), suite.chainA.SenderAccount.GetAddress(), ibctesting.TestCoin.Denom)).Sub(originalChainASenderAccountBalance[0]))
 }
+
+// TODO: determine if table tests are needed or not
+func (suite *FeeTestSuite) TestTransferFeeUpgrade() {
+	var path *ibctesting.Path
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {},
+			nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+
+			// configure the initial path to create a regular transfer channel
+			path.EndpointA.ChannelConfig.PortID = transfertypes.PortID
+			path.EndpointB.ChannelConfig.PortID = transfertypes.PortID
+			path.EndpointA.ChannelConfig.Version = transfertypes.Version
+			path.EndpointB.ChannelConfig.Version = transfertypes.Version
+
+			suite.coordinator.Setup(path)
+
+			// configure the channel upgrade to upgrade to an incentivized fee enabled transfer channel
+			upgradeVersion := string(types.ModuleCdc.MustMarshalJSON(&types.Metadata{FeeVersion: types.Version, AppVersion: transfertypes.Version}))
+			path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = upgradeVersion
+			path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = upgradeVersion
+
+			tc.malleate()
+
+			err := path.EndpointA.ChanUpgradeInit()
+			suite.Require().NoError(err)
+
+			err = path.EndpointB.ChanUpgradeTry()
+			suite.Require().NoError(err)
+
+			err = path.EndpointA.ChanUpgradeAck()
+			suite.Require().NoError(err)
```

Comment:
Needs https://github.com/cosmos/ibc-go/pull/4075 to move to OPEN without having to call another handler

Reply:
Thanks @DimitrisJim